src/main/java/org/elasticsearch/index/query/IdsQueryParser.java

```diff
@@ -107,26 +98,13 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
-
-        if (!idsProvided) {
-            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
-        }
-
         if (ids.isEmpty()) {
-            return Queries.newMatchNoDocsQuery();
-        }
-
-        if (types == null || types.isEmpty()) {
-            types = parseContext.queryTypes();
-        } else if (types.size() == 1 && Iterables.getFirst(types, null).equals("_all")) {
-            types = parseContext.mapperService().types();
-        }
-
-        TermsQuery query = new TermsQuery(UidFieldMapper.NAME, Uid.createTypeUids(types, ids));
-        query.setBoost(boost);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
+            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
```

Comment:
> I wonder if this distiction makes sense

testing is the way to find out, makes no sense to me either...

> I think we should allow no ids

not sure to be honest, What I care about is aligning validation between rest layer and java api. If we accept empty list let's accept it everywhere.